hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/utils/TestRowDataToAvroConverters.java

```diff
@@ -0,0 +1,124 @@
+
+import static org.apache.flink.table.api.DataTypes.ROW;
+import static org.apache.flink.table.api.DataTypes.FIELD;
+import static org.apache.flink.table.api.DataTypes.TIMESTAMP;
+
+class TestRowDataToAvroConverters {
+
+  DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+  @Test
+  void testRowDataToAvroStringToRowDataWithLocalTimezone1() throws JsonProcessingException {
+    TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("Asia/Shanghai")));
+    String timestampFromUtc8 = "2021-03-30 15:44:29";
+
+    DataType rowDataType = ROW(FIELD("timestamp_from_utc_8", TIMESTAMP()));
+    JsonToRowDataConverters.JsonToRowDataConverter jsonToRowDataConverter =
+            new JsonToRowDataConverters(true, true, TimestampFormat.SQL)
+            .createConverter(rowDataType.getLogicalType());
+    Object rowData = jsonToRowDataConverter.convert(new ObjectMapper().readTree("{\"timestamp_from_utc_8\":\"" + timestampFromUtc8 + "\"}"));
+
+    RowType rowType = (RowType) DataTypes.ROW(DataTypes.FIELD("f_timestamp", DataTypes.TIMESTAMP(3))).getLogicalType();
+    RowDataToAvroConverters.RowDataToAvroConverter converter =
+            RowDataToAvroConverters.createConverter(rowType, false);
+    GenericRecord avroRecord =
+            (GenericRecord) converter.convert(AvroSchemaConverter.convertToSchema(rowType), rowData);
+    Assertions.assertEquals(timestampFromUtc8, formatter.format(LocalDateTime.ofInstant(Instant.ofEpochMilli((Long) avroRecord.get(0)), ZoneId.systemDefault())));
+  }
+
+  @Test
+  void testRowDataToAvroStringToRowDataWithLocalTimezone2() throws JsonProcessingException {
+    TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("UTC+1")));
+    String timestampFromUtc1 = "2021-03-30 08:44:29";
+
+    DataType rowDataType = ROW(FIELD("timestamp_from_utc_1", TIMESTAMP()));
+    JsonToRowDataConverters.JsonToRowDataConverter jsonToRowDataConverter =
+            new JsonToRowDataConverters(true, true, TimestampFormat.SQL)
+                    .createConverter(rowDataType.getLogicalType());
+    Object rowData = jsonToRowDataConverter.convert(new ObjectMapper().readTree("{\"timestamp_from_utc_1\":\"" + timestampFromUtc1 + "\"}"));
+
+    RowType rowType = (RowType) DataTypes.ROW(DataTypes.FIELD("f_timestamp", DataTypes.TIMESTAMP(3))).getLogicalType();
+    RowDataToAvroConverters.RowDataToAvroConverter converter =
+            RowDataToAvroConverters.createConverter(rowType, false);
+    GenericRecord avroRecord =
+            (GenericRecord) converter.convert(AvroSchemaConverter.convertToSchema(rowType), rowData);
+    Assertions.assertEquals(timestampFromUtc1, formatter.format(LocalDateTime.ofInstant(Instant.ofEpochMilli((Long) avroRecord.get(0)), ZoneId.systemDefault())));
+  }
+
+  @Test
+  void testRowDataToAvroStringToRowDataWithLocalTimezone3() throws JsonProcessingException {
+    TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("UTC")));
+    String timestampFromUtc0 = "2021-03-30 07:44:29";
```

Comment:
should we set up the JVM timezone or just fetch the local timezone and check the result with the cosresponding timezone offset.

Reply:
Yes. That's a better way. I will refine UT and IT.

Reply:
Refinement is done.